proto/message_contents/content.proto

```diff
@@ -41,6 +41,8 @@ message EncodedContent {
     optional Compression compression = 5;
     // encoded content itself
     bytes content = 4;
+    // optional ID of a message that this message is referencing
+    optional string referencing_message_id = 6;
```

Comment:
Overall feels okay to me, but admittedly without thinking too hard about any far-reaching consequences. If `referencing_message_id` is the sha256 digest of the referenced message, this seems pretty safe. It's effectively a handle to aid in looking up the referenced message which should be very difficult to predict or forge (envelope contents include randomly generated salt, etc).

For tactical use seems totally fine.

Reply:
For my sake, might be good to reiterate in comment what the message id is e.g. "sha256 of message envelope content as unencoded bytes"